### ISSUE TITLE
Add support for markdown "checkboxes"

### ISSUE THREAD

**Issue by evolutionxbox** [Kilian/fromscratch#36]
When submitting an issue here on github, I am able to make lists with "checkboxes" which can be checked/unchecked.

For example, using:

```
- [ ] an example issue
- [x] a completed issue
```

will output:

- [ ] an example issue
- [x] a completed issue

---

Can a similar feature be added (without the fancy rendering)? 
Where one can insert a "checkbox" using a keyboard shortcut, and then check/uncheck it using another?

---

Example screenshot:

![screen shot 2016-11-01 at 15 53 45](https://cloud.githubusercontent.com/assets/660368/19896202/6059253e-a04b-11e6-8d5c-f9e1ddec514e.png)

**Comment by Kilian** [comment]
I've had this on my personal list as well! I like it a lot but haven't looked into it yet as it's changing text content. If it's doable in a non-hacky way, I'll try and add it.

**Comment by Kilian** [comment]
Any suggestions for the keyboard shortcut(s)?

**Comment by evolutionxbox** [comment]
Maybe something like <kbd>cmd</kbd> + <kbd>.</kbd> or <kbd>cmd</kbd> + <kbd>/</kbd>?

**Comment by Kilian** [comment]
I like the idea of <kbd>spacebar</kbd> to toggle, that keeps it consistent with html checkboxes. however that should only work when you are between [ ]. Would you expect <kbd>cmd</kbd>+<kbd>.</kbd> to work wherever you are on a line?

**Comment by Kilian** [comment]
Just pushed this to the checkbox-support branch. if you press <kbd>ctrl + /</kbd>, it will add an empty checkbox to the beginning of the current line (prepended by any whitespace you already have, so indented lists work) . pressing <kbd>ctrl + /</kbd> again will toggle the checkbox to checked and unchecked.

![checkboxes](https://cloud.githubusercontent.com/assets/41970/20077357/59ef133e-a53c-11e6-936d-ff248eb604ea.gif)

What this doesn't do is append the checkbox after any "list" items such as * or -, it will instead add the checkbox *before* the list item but *after* any whitespace at the beginning of the line. To me this looks better (and it's less code to write and maintain, so double win). Let me know what you think @evolutionxbox!

**Comment by evolutionxbox** [comment]
I really like this. I'm going to checkout that branch later today and have a play around. Personally I think you've got a good approach.

**Comment by Kilian** [comment]
Just pushed support for interaction with selections too: select multiple lines and press ctrl + / and it will add/toggle checkboxes on all of them.

**Comment by nkantar** [comment]
I'm using FromScratch as my to-do scratchpad (it feels like a much nicer macOS Stickies), and I'm really looking forward to this in stable. 👍 

Supporting  list-style checkboxes may be useful for those who want Markdown in the end, but I think I can live without them.

**Comment by Kilian** [comment]
What are list-style checkboxes?

**Comment by nkantar** [comment]
The GitHub flavored Markdown syntax of:

```
- [ ] one
- [ ] two
    - [ ] three
- [x] four
- [ ] five
```

...which they render as so:

- [ ] one
- [ ] two
    - [ ] three
- [x] four
- [ ] five

**Comment by Kilian** [comment]
Ah, yes. We're not going to render checkboxes as html, they will remain plain text. IMO the dash is superfluous:
```
[ ] one
[ ] two
    [ ] two point one
[ ] three
```
Reads just as well, right? See my comment here for rationale: https://github.com/Kilian/fromscratch/issues/36#issuecomment-258973814

**Comment by evolutionxbox** [comment]
@Kilian yeah I agree with you. The checkboxes do not need to render.

**Comment by nkantar** [comment]
Yes, it _reads_ fine, but isn't great for copy/pasting into Markdown files.

And just to be clear, I don't think FromScratch should _render_ them as HTML — I'd just personally prefer `- [ ] one` over `[ ] one`.

That said, it's perfectly fine for FromScratch not to bother with that, as it's not a Markdown editor (and doesn't want to be one). If I end up needing something like this, processing text is easy enough to script.

**Comment by Kilian** [comment]
Just merged this back to master, will be in 1.3.0!
